opennms-model/src/main/java/org/opennms/netmgt/model/OnmsSnmpInterface.java

```diff
@@ -766,10 +805,8 @@ public void mergeSnmpInterfaceAttributes(OnmsSnmpInterface scannedSnmpIface) {
             setLastSnmpPoll(scannedSnmpIface.getLastSnmpPoll());
         }
         
-        if(scannedSnmpIface.isCollectionUserSpecified() || !isCollectionUserSpecified()){
-            setCollectionEnabled(scannedSnmpIface.isCollectionEnabled(), scannedSnmpIface.isCollectionUserSpecified());
-        }
-        
+        setCollectionEnabled(scannedSnmpIface.isCollectionEnabled(), scannedSnmpIface.getCollectionDefinitionSource());
```

Comment:
Shouldn't this be 
```suggestion
        setCollectionEnabled(scannedSnmpIface.isCollectionEnabled(), getCollectionDefinitionSource());
```

Reply:
I don't think so. I changed the behavior of `setCollectionEnabled` so that it treats `source` as the scanned interface's source. It compares the incoming source against its own internal source, and only allows overrides if the incoming source takes priority over (or matches) its internal one.

Reply:
yeah, I understand it now after looking it one more time.  Nice work!  This should fix the bug as per my understanding.